src/ui/backend/BaseBacking.js

```diff
@@ -62,36 +83,39 @@ export default class BaseBacking {
 
   copy () {
     var copy = {};
+    for (var key in BASE_STYLE_PROPS) {
+      copy[key] = this[key];
+    }
 
-    for (var i = 0; i < styleKeyList.length; i++) {
-      var key = styleKeyList[i];
+    for (var i = 0; i < extendedStylePropList.length; i++) {
+      var key = extendedStylePropList[i];
       copy[key] = this[key];
     }
 
     return copy;
   }
 
   update (style) {
-    for (var i = 0; i < styleKeyList.length; i++) {
-      var key = styleKeyList[i];
+    for (var key in style) {
+      if (this[key] !== undefined) {
+        this[key] = style[key];
+      }
+    }
+
+    for (var i = 0; i < extendedStylePropList.length; i++) {
+      var key = extendedStylePropList[i];
       if (style[key] !== void 0) {
         this[key] = style[key];
       }
     }
+
     return this;
   }
 
 };
 
-var styleKeys = BaseBacking.prototype.constructor.styleKeys = {};
-var styleKeyList = [];
-
+var extendedStylePropList = [];
```

Comment:
Currently devkit only has one extension, the layout system, added to views, but it's added to all views regardless of whether or not they use it. The number of extended properties is greater than the default properties too. I'm wondering if adding the extension to only the views that need it would be a further improvement. We could make the boolean `layout` property a getter / setter, and dynamically add the layout extension as needed. None of our gameplay views or views in particle engines would ever need the layout properties. See [BackingExtension.js](https://github.com/gameclosure/timestep/blob/es6/src/ui/layout/BackingExtension.js).